Scripts/Extension/GameObjectExtensions.cs

```diff
@@ -8,13 +8,14 @@
     public static class GameObjectExtensions
     {
         /// <summary>
-        /// Attempts to retrieve the <see cref="Component"/> from a given <see cref="GameObject"/>.
+        /// Attempts to retrieve the component from a given <see cref="GameObject"/>.
         /// </summary>
-        /// <param name="gameObject">The <see cref="GameObject"/> to retrieve the <see cref="Component"/> from.</param>
-        /// <returns>The <see cref="Component"/> if one exists on the given <see cref="GameObject"/>.</returns>
-        public static Component TryGetComponent(this GameObject gameObject)
+        /// <typeparam name="T">The component type to retrieve.</typeparam>
+        /// <param name="gameObject">The <see cref="GameObject"/> to retrieve from.</param>
+        /// <returns>The component if one exists.</returns>
+        public static T TryGetComponent<T>(this GameObject gameObject)

```

Comment:
Missing type constraint:
```c#
public static T TryGetComponent<T>(this GameObject gameObject) where T : Component
```

Reply:
Isn't there a thing where `GetComponent` can also get things that don't derive from component? Like Renderers and stuff like that?

Reply:
They all inherit from `Component`. Our issue was that a `Renderer` for example doesn't inherit from _`Behaviour`_.